File: tests/test-find-new.pl
use strict;
use warnings;
use Encode;
use Selenium::Remote::Driver;
use Test::More;

my $driver = Selenium::Remote::Driver->new( remote_server_addr => "localhost",
                                               port => 4444,
                                               browser_name => "firefox");

$driver->get("http://credoweb.bg");
$driver->find_element(Encode::decode('UTF-8', "span.element-heading"), "css")->click;
$driver->find_element(Encode::decode('UTF-8', "city"), "name")->click;
$driver->find_element(Encode::decode('UTF-8', "city"), "name")->clear;
$driver->find_element(Encode::decode('UTF-8', "city"), "name")->send_keys("София");
$driver->find_element(Encode::decode('UTF-8', "//input[\@value='Търси']"), "xpath")->click;
$driver->find_element(Encode::decode('UTF-8', "Димитър Христев"), "link")->click;
$driver->find_element(Encode::decode('UTF-8', "ЗА МЕН"), "link")->click;
$driver->find_element(Encode::decode('UTF-8', "ВСИЧКО"), "link")->click;
$driver->quit();
done_testing();

File: tests/test-find-new.java
package com.example.tests;

import java.util.regex.Pattern;
import java.util.concurrent.TimeUnit;
import org.junit.*;
import static org.junit.Assert.*;
import static org.hamcrest.CoreMatchers.*;
import org.openqa.selenium.*;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

public class TestFindNew {
  private WebDriver driver;
  private String baseUrl;
  private boolean acceptNextAlert = true;
  private StringBuffer verificationErrors = new StringBuffer();

  @Before
  public void setUp() throws Exception {
    driver = new FirefoxDriver();
    baseUrl = "http://www.credoweb.bg/";
    driver.manage().timeouts().implicitlyWait(30, TimeUnit.SECONDS);
  }

  @Test
  public void testFindNew() throws Exception {
    driver.get(baseUrl + "/");
    driver.findElement(By.cssSelector("span.element-heading")).click();
    driver.findElement(By.name("city")).click();
    driver.findElement(By.name("city")).clear();
    driver.findElement(By.name("city")).sendKeys("София");
    driver.findElement(By.xpath("//input[@value='Търси']")).click();
    driver.findElement(By.linkText("Димитър Христев")).click();
    driver.findElement(By.linkText("За мен")).click();
    driver.findElement(By.linkText("Всичко")).click();
  }

  @After
  public void tearDown() throws Exception {
    driver.quit();
    String verificationErrorString = verificationErrors.toString();
    if (!"".equals(verificationErrorString)) {
      fail(verificationErrorString);
    }
  }

  private boolean isElementPresent(By by) {
    try {
      driver.findElement(by);
      return true;
    } catch (NoSuchElementException e) {
      return false;
    }
  }

  private boolean isAlertPresent() {
    try {
      driver.switchTo().alert();
      return true;
    } catch (NoAlertPresentException e) {
      return false;
    }
  }

  private String closeAlertAndGetItsText() {
    try {
      Alert alert = driver.switchTo().alert();
      String alertText = alert.getText();
      if (acceptNextAlert) {
        alert.accept();
      } else {
        alert.dismiss();
      }
      return alertText;
    } finally {
      acceptNextAlert = true;
    }
  }
}
